system/jlib/jsocket.cpp

```diff
@@ -5450,13 +5504,24 @@ class CSocketEpollThread: public CSocketBaseThread
         CriticalBlock block(sect);
         if (terminating)
             return 0;
-        unsigned rm = 0;
-        if (removeSock(sock))
-            rm = SOCK_REMOVED;
-        unsigned n = items.ordinality();
+
+        // JCSMORE may be more efficient to use a double linked list to avoid this scan (and array/expansion)
+        unsigned n=0;
+        ForEachItemIn(i, items)
+        {
+            SelectItem &si = *items.element(i);
+            if (!si.del)
+            {
+                if (si.sock==sock)
+                    si.del = true;
+                else
+                    n++;
+            }
+        }
+
         // new handler thread
         if (n >= hdlPerThrd)
-            return (0|rm);
+            return false;
```

Comment:
more: The previous logic could mean that you could avoid walking any further handlers once the socket had been added and removed.  I don't know how commit it is to replace rather than add fresh.

Reply:
right, except, due to a bug where called, it would exit if only removed (in practice as noted in the PR comment, this would never actually happen since capacity is UINT_MAX).

But, yes, because of the bug, I missed the purpose of the logic, I'll restore and fix bug, i.e.:
```
                if ((addrm & (SOCK_ADDED | SOCK_REMOVED)) == (SOCK_ADDED | SOCK_REMOVED))
```